bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -0,0 +1,348 @@
+    if (useCachedDataPoolStr != null) {
+      builder.setUseCachedDataPool(Boolean.parseBoolean(useCachedDataPoolStr));
+    }
+
+    // This information is in addition to bigtable-client-core version, and jdk version.
+    StringBuilder agentBuilder = new StringBuilder();
+    agentBuilder.append("hbase-").append(VersionInfo.getVersion());
+    String customUserAgent = configuration.get(CUSTOM_USER_AGENT_KEY);
+    if (customUserAgent != null) {
+      agentBuilder.append(',').append(customUserAgent);
+    }
+    builder.setUserAgent(agentBuilder.toString());
+  }
+
+  private void setBulkOptions(BigtableOptions.Builder bigtableOptionsBuilder) {
+    BulkOptions.Builder bulkOptionsBuilder = BulkOptions.builder();
+
+    int asyncMutatorCount =
+        configuration.getInt(
+            BIGTABLE_ASYNC_MUTATOR_COUNT_KEY, BIGTABLE_ASYNC_MUTATOR_COUNT_DEFAULT);
+    bulkOptionsBuilder.setAsyncMutatorWorkerCount(asyncMutatorCount);
+
+    bulkOptionsBuilder.setUseBulkApi(configuration.getBoolean(BIGTABLE_USE_BULK_API, true));
+    bulkOptionsBuilder.setBulkMaxRowKeyCount(
+        configuration.getInt(
+            BIGTABLE_BULK_MAX_ROW_KEY_COUNT, BIGTABLE_BULK_MAX_ROW_KEY_COUNT_DEFAULT));
+    bulkOptionsBuilder.setBulkMaxRequestSize(
+        configuration.getLong(
+            BIGTABLE_BULK_MAX_REQUEST_SIZE_BYTES, BIGTABLE_BULK_MAX_REQUEST_SIZE_BYTES_DEFAULT));
+    bulkOptionsBuilder.setAutoflushMs(
+        configuration.getLong(BIGTABLE_BULK_AUTOFLUSH_MS_KEY, BIGTABLE_BULK_AUTOFLUSH_MS_DEFAULT));
+
+    int defaultRpcCount =
+        BIGTABLE_MAX_INFLIGHT_RPCS_PER_CHANNEL_DEFAULT
+            * bigtableOptionsBuilder.getDataChannelCount();
+    int maxInflightRpcs = configuration.getInt(MAX_INFLIGHT_RPCS_KEY, defaultRpcCount);
+    bulkOptionsBuilder.setMaxInflightRpcs(maxInflightRpcs);
+
+    long maxMemory =
+        configuration.getLong(
+            BIGTABLE_BUFFERED_MUTATOR_MAX_MEMORY_KEY, BulkOptions.BIGTABLE_MAX_MEMORY_DEFAULT);
+    bulkOptionsBuilder.setMaxMemory(maxMemory);
+
+    if (configuration.getBoolean(
+        BIGTABLE_BUFFERED_MUTATOR_ENABLE_THROTTLING,
+        BulkOptions.BIGTABLE_BULK_ENABLE_THROTTLE_REBALANCE_DEFAULT)) {
+      LOG.info(
+          "Bigtable mutation latency throttling enabled with threshold %d",
+          configuration.getInt(
+              BIGTABLE_BUFFERED_MUTATOR_THROTTLING_THRESHOLD_MILLIS,
+              BulkOptions.BIGTABLE_BULK_THROTTLE_TARGET_MS_DEFAULT));
+      bulkOptionsBuilder.enableBulkMutationThrottling();
+      bulkOptionsBuilder.setBulkMutationRpcTargetMs(
+          configuration.getInt(
+              BIGTABLE_BUFFERED_MUTATOR_THROTTLING_THRESHOLD_MILLIS,
+              BulkOptions.BIGTABLE_BULK_THROTTLE_TARGET_MS_DEFAULT));
+    }
+
+    bigtableOptionsBuilder.setBulkOptions(bulkOptionsBuilder.build());
+  }
+
+  private void setCredentialOptions(BigtableOptions.Builder builder) throws FileNotFoundException {
+    if (configuration.getBoolean(
```

Comment:
This is important for directpath: we should not overwrite the default credentials if the user has not explicitly set an override

Reply:
I am assuming user could override credentials through `BigtableExtendedConfiguration#withCredentials` OR Service Account(JSON file/Email & P12 key location). If neither of them presents then use the default one.